pkg/render/compliance.go

```diff
@@ -751,14 +763,13 @@ func complianceVolumes(defaultMode int32, managerSecret *corev1.Secret) []corev1
 			},
 		}}}
 
-	if managerSecret != nil {
+	if c.managerInternalTLSSecret != nil {
 		volumes = append(volumes,
 			corev1.Volume{
 				Name: ManagerInternalTLSSecretName,
 				VolumeSource: corev1.VolumeSource{
 					Secret: &corev1.SecretVolumeSource{
-						DefaultMode: &defaultMode,
```

Comment:
And this?

Reply:
It is the k8s default and therefore redundant.

Reply:
Redundant isn't necessarily a bad thing - I'd rather be explicit and verbose rather than implicit and harder to understand. 

I'd suggest, if it isn't required for this functionality, let's keep it out of this PR. Makes it easier to focus on the important changes.